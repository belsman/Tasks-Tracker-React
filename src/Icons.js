import React from 'react';
import { faHourglass, faPlus, faChartLine, faChartPie, faEllipsisH, faAngleRight } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const AddMeasurementIcon = () => (
  <span>
    <FontAwesomeIcon icon={faHourglass} size="lg"/>
    <sup><FontAwesomeIcon icon={faPlus} size="sm"/></sup>
  </span>
);

export const ListMeasurementIcon = () => (
    <span>
      <FontAwesomeIcon icon={faChartLine} size="lg"/>
    </span>
);

export const ProgressIcon = () => (
    <span>
      <FontAwesomeIcon icon={faChartPie} size="lg"/>
    </span>
);

export const MoreIcon = () => (
    <span>
      <FontAwesomeIcon icon={faEllipsisH} size="lg"/>
    </span>
);

export const RightIcon = () => (
  <span>
    <FontAwesomeIcon icon={faAngleRight} size="md"/>
  </span>
);
