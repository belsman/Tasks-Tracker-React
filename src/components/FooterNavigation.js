import React from 'react';
import { NavLink } from 'react-router-dom';
import {
  AddMeasurementIcon, ListMeasurementIcon, ProgressIcon, MoreIcon,
} from './Icons';

import style from './footerNav.module.css';

const FooterNavigation = () => {
  const checkActiveLink = path => (match, location) => location.pathname === path;

  return (
    <nav className={style.nav}>
      <div className={style.navLinks}>
        <NavLink
          to="/"
          activeClassName="navActive"
          isActive={checkActiveLink('/')}
        >
          <AddMeasurementIcon />
        </NavLink>

        <NavLink
          to="/records"
          activeClassName="navActive"
          isActive={checkActiveLink('/records')}
        >
          <ListMeasurementIcon />
        </NavLink>

        <NavLink to="#">
          <ProgressIcon />
        </NavLink>

        <NavLink to="#">
          <MoreIcon />
        </NavLink>
      </div>
    </nav>
  );
};

export default FooterNavigation;
