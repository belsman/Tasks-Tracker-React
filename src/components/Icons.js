import React from 'react';
import {
  faHourglass, faPlus, faChartLine, faChartPie, faEllipsisH,
  faAngleRight, faAngleLeft, faTerminal, faRunning,
  faBookOpen, faTools, faFilm, faSignOutAlt,
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const AddMeasurementIcon = () => (
  <span>
    <FontAwesomeIcon icon={faHourglass} size="lg" />
    <sup><FontAwesomeIcon icon={faPlus} size="sm" /></sup>
  </span>
);

export const ListMeasurementIcon = () => (
  <span>
    <FontAwesomeIcon icon={faChartLine} size="lg" />
  </span>
);

export const ProgressIcon = () => (
  <span>
    <FontAwesomeIcon icon={faChartPie} size="lg" />
  </span>
);

export const MoreIcon = () => (
  <span>
    <FontAwesomeIcon icon={faEllipsisH} size="lg" />
  </span>
);

export const LogOutIcon = () => (
  <span>
    <FontAwesomeIcon icon={faSignOutAlt} size="lg" />
  </span>
);

export const RightIcon = () => (
  <span>
    <FontAwesomeIcon icon={faAngleRight} size="md" />
  </span>
);

export const LeftIcon = () => (
  <span>
    <FontAwesomeIcon icon={faAngleLeft} size="md" />
  </span>
);

export const CodingIcon = () => (
  <span>
    <FontAwesomeIcon icon={faTerminal} size="lg" />
  </span>
);

export const RunningIcon = () => (
  <span>
    <FontAwesomeIcon icon={faRunning} size="lg" />
  </span>
);

export const ReadingIcon = () => (
  <span>
    <FontAwesomeIcon icon={faBookOpen} size="lg" />
  </span>
);

export const MovieIcon = () => (
  <span>
    <FontAwesomeIcon icon={faFilm} size="lg" />
  </span>
);

export const ProjectIcon = () => (
  <span>
    <FontAwesomeIcon icon={faTools} size="lg" />
  </span>
);
