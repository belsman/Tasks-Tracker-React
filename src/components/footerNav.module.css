.nav {
  position: fixed;
  width: 100%;
  bottom: 0;
  overflow: hidden;
  background-color: #212529;
}

.navLinks {
  display: flex;
  justify-content: space-between;
}

.navLinks a {
  display: block;
  flex-basis: 25%;
  text-align: center;
  color: #fff;
  padding: 0.5rem;
  text-decoration: none;
}

.navLinks a:hover {
  background-color: #42b5e8;
}
