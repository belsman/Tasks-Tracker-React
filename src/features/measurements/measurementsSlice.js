import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import client from '../../app/server';

const initialState = {
  measurements: [],
  status: 'idle',
  addMeasurementStatus: 'idle',
  error: null,
};

export const fetchMeasurements = createAsyncThunk('measurements/fetchMeasurement', async options => {
  const { token } = options;
  const response = await client.get('/measurements', token);
  const data = await response.json();
  return data;
});

export const addNewMeasurement = createAsyncThunk('measurements/addNewMeasurement', async options => {
  const { token, formData } = options;
  const response = await client.post('/measurements', token, formData);
  const data = await response.json();
  return data;
});

const measurementsSlice = createSlice({
  name: 'measurements',
  initialState,
  reducers: {},
  extraReducers: {
    /* eslint-disable no-param-reassign */
    [fetchMeasurements.pending]: state => {
      state.status = 'loading';
    },
    [fetchMeasurements.fulfilled]: (state, action) => {
      state.status = 'succeeded';
      state.measurements = state.measurements.concat(action.payload);
    },
    [fetchMeasurements.rejected]: (state, action) => {
      state.status = 'failed';
      state.error = action.error.message;
    },
    [addNewMeasurement.pending]: (state, action) => {
      state.addMeasurementStatus = 'loading';
      state.measurements.push(action.payload);
    },
    [addNewMeasurement.fulfilled]: (state, action) => {
      state.measurements.push(action.payload);
      state.addMeasurementStatus = 'idle';
    },
    /* eslint-enable no-param-reassign */
  },
});

export default measurementsSlice.reducer;

export const selectAllMeasurements = state => state.measurements.measurements;

export const selectMeasurementsById = (state, measurementId) => (
  state.measurements.measurements.filter(measured => measured !== null)
    .find(measured => String(measured.id) === measurementId)
);
