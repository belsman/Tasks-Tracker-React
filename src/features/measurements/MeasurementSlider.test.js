import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import pretty from 'pretty';
import { act } from 'react-dom/test-utils';

/* eslint-disable no-unused-vars */
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { format } from 'date-fns';
import { RightIcon, LeftIcon } from '../../components/Icons';
import style from './singleRecord.module.css';
import { selectAllMeasurements } from './measurementsSlice';
/* eslint-enable no-unused-vars */

import Slider from './MeasurementSlider';

jest.mock('./measurementsSlice', () => ({
  selectAllMeasurements(fn) {
    return fn;
  },
}));

jest.mock('react-router-dom', () => ({
  useHistory() {
    return [];
  },
}));

jest.mock('react-redux', () => ({
  useSelector() {
    return [
      {
        id: 1,
        running: 1,
        reading: 2,
        project: 3,
        coding: 4,
        movie: 5,
        created_at: '2013-03-10T02:00:00Z',
      },
      {
        id: 2,
        running: 1,
        reading: 2,
        project: 3,
        coding: 4,
        movie: 5,
        created_at: '2013-03-10T02:00:00Z',
      },
    ];
  },
}));

jest.mock('date-fns', () => ({
  format() {
    return (
      'March 13th, 1962'
    );
  },
}));

jest.mock('../../components/Icons', () => ({
  RightIcon() {
    return (
      <div>An Icon</div>
    );
  },
  LeftIcon() {
    return (
      <div>An Icon</div>
    );
  },
}));

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  act(() => {
    render(<Slider record="1223445677" recordId={1} />, container);
  });
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders with with the correct timestamp for a given the record', () => {
  const slideBtns = document.querySelectorAll('button');
  expect(slideBtns).toHaveLength(2);
});

it('should render the expected output', () => {
  act(() => {
    render(<Slider record="1223445677" recordId={1} />, container);
  });

  expect(
    pretty(container.innerHTML),
  ).toMatchSnapshot();
});
