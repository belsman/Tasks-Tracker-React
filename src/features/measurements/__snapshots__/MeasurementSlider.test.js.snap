// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expected output 1`] = `
"<div class=\\"slider\\"><button type=\\"button\\" disabled=\\"\\">
    <div>An Icon</div>
  </button><span>March 13th, 1962</span><button type=\\"button\\">
    <div>An Icon</div>
  </button></div>"
`;
